From 88200ca1eaff7a64100edca55f7f3f6070935cf4 Mon Sep 17 00:00:00 2001
From: wh <wanghang@rock-chips.com>
Date: Fri, 18 Oct 2019 19:47:04 +0800
Subject: [PATCH 199/297] services: grant provider permission to allow install
 apk

add com.android.rk.fileprovider uri permission for RKExplorer

Change-Id: I6efd0b4de4e6438e832bd6c53cf88d853fb9ad9f
Signed-off-by: wh <wanghang@rock-chips.com>
(cherry picked from commit 823191035f6f088fd6a0a254a1768559be90b213)
---
 .../java/com/android/server/uri/UriGrantsManagerService.java     | 1 +
 1 file changed, 1 insertion(+)

diff --git a/services/core/java/com/android/server/uri/UriGrantsManagerService.java b/services/core/java/com/android/server/uri/UriGrantsManagerService.java
index f14c3a53940..2fccfdac758 100644
--- a/services/core/java/com/android/server/uri/UriGrantsManagerService.java
+++ b/services/core/java/com/android/server/uri/UriGrantsManagerService.java
@@ -1085,6 +1085,7 @@ public class UriGrantsManagerService extends IUriGrantsManager.Stub {
         final int callingAppId = UserHandle.getAppId(callingUid);
         if ((callingAppId == SYSTEM_UID) || (callingAppId == ROOT_UID)) {
             if ("com.android.settings.files".equals(grantUri.uri.getAuthority())
+                    || "com.android.rk.fileprovider".equals(grantUri.uri.getAuthority())
                     || "com.android.settings.module_licenses".equals(grantUri.uri.getAuthority())) {
                 // Exempted authority for
                 // 1. cropping user photos and sharing a generated license html
-- 
2.36.0

